nixos/modules/services/backup/borgbackup.nix

```
@@ -284,7 +284,7 @@ in {
           };
 
           archiveBaseName = mkOption {
-            type = types.strMatching "[^/{}]+";
+            type = types.nullOr (types.strMatching "[^/{}]+");
```

this might be easier overall since you coalesce `null` to `""` anyway:
```suggestion
            type = types.strMatching "[^/{}]*";
```